micrometer-core/src/main/java/io/micrometer/core/instrument/Meter.java

```diff
@@ -480,6 +481,26 @@ public Meter register(MeterRegistry registry) {
 
     }
 
+    /**
+     * Convenience interface to create new meters from tags based on a common
+     * "template"/builder. See usage in Meter implementations, e.g.: {@code Timer},
+     * {@code Counter}
+     *
+     * @param <T> Meter type
+     * @since 1.12.0
+     */
+    interface MeterProvider<T extends Meter> {
+
+        /**
+         * Registers (creates a new or gets an existing one if already exists) Meters
+         * using the provided tags.
+         * @param tags Tags to attach to the Meter about to be registered
+         * @return A new or existing Meter
+         */
+        T withTags(Iterable<? extends Tag> tags);
```

Comment:
Can this also support overloaded methods for `String... tags` and `String key, String value` like most metrics have?

Reply:
Yes, I think we can add default implementations for those that delegate to this method for convenience.

Reply:
👍🏼 I added them.